libibverbs/driver.h

```diff
@@ -102,6 +102,15 @@ struct verbs_qp {
 };
 static_assert(offsetof(struct ibv_qp_ex, qp_base) == 0, "Invalid qp layout");
 
+struct verbs_cq {
+	union {
+		struct ibv_cq cq;
+		struct ibv_cq_ex cq_ex;
+	};
+};
```

Comment:
What is the use case that you come to solve with this patch ? is it only some preparation for future work ? we usually prefer having some series with some added value / usage and not only preparations.

Reply:
This is a cleanup patch, preventing the need for explicit casts. The use case is the providers supporting extended CQ, mlx4/5 and any future providers.